### PR TITLE
Ensure resource directories exist in jars

### PR DESCRIPTION
**Pull request by restingbull**
Fixes https://github.com/bazelbuild/rules_kotlin/issues/582

### CODE REVIEW

File: kotlin/internal/jvm/compile.bzl [hunk 1]
```diff
@@ -175,6 +175,7 @@ def _fold_jars_action(ctx, rule_kind, toolchains, output_jar, input_jars, action
         "--normalize",
         "--compression",
         "--exclude_build_data",
+        "--add_missing_directories",
```
Comment (cgruber) [inline kotlin/internal/jvm/compile.bzl:178]: Man, I'm confused about this PR... is the fundamental fix adding this flag, or adding the closeable that forces execution? Or is the closeable just do you can use it more conveniently in tests in a `try-with-resources`/`Closeable.use{}`? 

Anyway, I'd laugh/cry if this flag were the fix...  Sigh.

**Review by cgruber — APPROVED** [review]

**Comment by cgruber** [comment]
Note, I'll leave merging to you, since this clearly builds on #612

File: kotlin/internal/jvm/compile.bzl (hunk 1 above)
Comment (restingbull) [reply kotlin/internal/jvm/compile.bzl:178]: this flag is the fix, but I was adding tests to all of 'em.

The closeable just makes the code read cleaner.

**Comment by restingbull** [comment]
> Note, I'll leave merging to you, since this clearly builds on #612

Unintentionally.

**Pull request merged**